theano/gof/opt.py

```diff
@@ -1261,10 +1289,78 @@ def tracks(self):
+            opts = self.track_map[type(node.op)] + self.track_map[node.op] + self.track_map[None]
+            repl = prev_repl
+            new_repl = None
+            for opt in opts:
+                opt_start = time.time()
+                repl = opt.transform(node)
+                opt_finish = time.time()
+                self.time_opts[opt] += opt_start - opt_finish
+                self.process_count[opt] += 1
+                if not repl:
+                    continue
+                else:
+                    self.node_created[opt] += len(graph.ops(fgraph.variables, repl))
+                    self.applied_true[opt] += 1
+                    if not multiple_opts or not repl[0].owner:
+                        return repl
+                    assert len(repl) == 1
+                    # Ensuring not the input of graph
+                    assert repl[0].owner
+                    new_node = repl[0].owner
+                    new_repl = apply_mult_opts(new_node, fgraph, True, repl)
+            if new_repl:
+                repl = new_repl
+            return repl
+
+        node_start = time.time()
+        new_var = apply_mult_opts(node, node.fgraph, self.apply_all_opts)
+        node_finish = time.time()
+        self.time_nodes[node] = node_finish - node_start
+        return new_var
+
+    @staticmethod
+    def print_profile(stream, prof, level=0):
+        (time_opts, time_nodes, process_count, applied_true, node_created) = prof
+        blanc = ('    ' * int(level))
+        print(blanc, "LocalOptGroup", file=stream)
+        print(blanc, "---------------------", file=stream)
+        count_opt = []
+        not_used = []
+        not_used_time = 0
+        for o, count in iteritems(process_count):
+            if count > 0:
+                count_opt.append((time_opts[o], applied_true[o], count, o, node_created[o]))
+            else:
+                not_used.append((time_opts[o], o))
+                not_used_time += time_opts[o]
+        if count_opt:
+            print(blanc,
+                  '  time taken - times applied - times tried - name - node_created:',
+                  file=stream)
+            count_opt.sort()
+            for (t, a_t, count, o, n_c) in count_opt[::-1]:
+                print(blanc, '  %.3fs - %d - %d - %s - %d' % (
+                      t, a_t, count, o, n_c), file=stream)
+            print(blanc, '  %.3fs - in %d optimization that were not used (display those with runtime greater than 0)' % (
+                not_used_time, len(not_used)), file=stream)
+            not_used.sort(key=lambda nu: (nu[0], str(nu[1])))
+            for (t, o) in not_used[::-1]:
+                if t > 0:
+                    # Skip opt that have 0 times, they probably wasn't even tried.
+                    print(blanc + "  ", '  %.3fs - %s' % (t, o), file=stream)
+        else:
+            print(blanc, "--- The Optimizer wasn't successful ---", file=stream)
```

Comment:
Change the way it is printed. In your profile, it looks ugly. Add an extra level to the blanc and remove the dash.